plugin/pkg/scheduler/factory/factory_test.go

```diff
@@ -48,13 +48,79 @@ func TestCreate(t *testing.T) {
 
 func TestPollMinions(t *testing.T) {
 	table := []struct {
-		minions []api.Node
+		minions       []api.Node
+		expectedCount int
 	}{
 		{
 			minions: []api.Node{
-				{ObjectMeta: api.ObjectMeta{Name: "foo"}},
-				{ObjectMeta: api.ObjectMeta{Name: "bar"}},
+				{
+					ObjectMeta: api.ObjectMeta{Name: "foo"},
+					Status: api.NodeStatus{
+						Conditions: []api.NodeCondition{
+							{Kind: api.NodeReady, Status: api.ConditionFull},
+						},
+					},
+				},
+				{
+					ObjectMeta: api.ObjectMeta{Name: "bar"},
+					Status: api.NodeStatus{
+						Conditions: []api.NodeCondition{
+							{Kind: api.NodeReachable, Status: api.ConditionFull},
+						},
+					},
+				},
+				{
+					ObjectMeta: api.ObjectMeta{Name: "baz"},
+					Status: api.NodeStatus{
+						Conditions: []api.NodeCondition{
+							{Kind: api.NodeReady, Status: api.ConditionFull},
+							{Kind: api.NodeReachable, Status: api.ConditionFull},
+						},
+					},
+				},
+				{
+					ObjectMeta: api.ObjectMeta{Name: "baz"},
+					Status: api.NodeStatus{
+						Conditions: []api.NodeCondition{
+							{Kind: api.NodeReady, Status: api.ConditionFull},
+							{Kind: api.NodeReady, Status: api.ConditionFull},
+						},
+					},
+				},
 			},
+			expectedCount: 4,
+		},
+		{
+			minions: []api.Node{
+				{
+					ObjectMeta: api.ObjectMeta{Name: "foo"},
+					Status: api.NodeStatus{
+						Conditions: []api.NodeCondition{
+							{Kind: api.NodeReady, Status: api.ConditionFull},
+						},
+					},
+				},
+				{
+					ObjectMeta: api.ObjectMeta{Name: "bar"},
+					Status: api.NodeStatus{
+						Conditions: []api.NodeCondition{
+							{Kind: api.NodeReady, Status: api.ConditionNone},
+						},
+					},
+				},
+			},
+			expectedCount: 1,
+		},
+		{
+			minions: []api.Node{
+				{
+					ObjectMeta: api.ObjectMeta{Name: "foo"},
+					Status: api.NodeStatus{
+						Conditions: []api.NodeCondition{},
+					},
+				},
+			},
+			expectedCount: 1,
```

Comment:
Please add test cases exercising these two things, either of them should turn up a bug:

```
  {Kind: api.NodeReady, Status: api.ConditionFull},
  {Kind: api.NodeReachable, Status: api.ConditionNone},
```

```
  {Kind: api.NodeReachable, Status: api.ConditionFull},
  {Kind: "invalidValue", Status: api.ConditionNone},
```